include/ginkgo/core/base/types.hpp

```diff
@@ -35,6 +35,22 @@
 #endif  // defined(__CUDACC__) || defined(__HIPCC__)
 
 
+// Macros for handling different device error return types uniformly
+#if defined(__CUDACC__)
+#define GKO_DEVICE_ERROR_TYPE cudaError_t
+#define GKO_DEVICE_ERROR_INVALID cudaErrorInvalidValue
+#define GKO_DEVICE_NO_ERROR cudaSuccess
+#elif defined(__HIPCC__)
+#define GKO_DEVICE_ERROR_TYPE hipError_t
+#define GKO_DEVICE_ERROR_INVALID hipErrorInvalidValue
+#define GKO_DEVICE_NO_ERROR hipSuccess
+#else
```

Comment:
Since only succeess/failure is needed, you don't need these wrappers.

Reply:
The plan is to use these for aliasing the shared memory pointers in the device kernels as well. In those cases, we would need the return code which would be useful in narrowing the issue.

Reply:
btw, does this have to be in a public header? Wouldn't a core header be sufficient?

Reply:
have the same question as MarcelKoch. the current usage only require these alias in the core header not public.